sgkit/stats/association.py

```diff
@@ -203,10 +211,11 @@ def gwas_linear_regression(
     Y = Y.rechunk((None, -1))
 
     res = linear_regression(G.T, X, Y)
-    return xr.Dataset(
+    new_ds = xr.Dataset(
         {
             "variant_beta": (("variants", "traits"), res.beta),
             "variant_t_value": (("variants", "traits"), res.t_value),
             "variant_p_value": (("variants", "traits"), res.p_value),
         }
     )
+    return merge_datasets(ds, new_ds) if merge else new_ds
```

Comment:
I wonder if it's worth having a function that does the conditional merge? I can see it being a bit error-prone having this logic in every function, maybe:
```python
return conditional_merge_datasets(ds, new_ds, merge)
```

Reply:
See https://github.com/pystatgen/sgkit/pull/250/commits/b634bd71dd70aaa3199399bf6e2c179c9eb4e6d2